recipe/meta.yaml

```diff
@@ -63,17 +68,25 @@ outputs:
         - llvm-openmp  # [osx]
         - libgomp      # [linux]
       host:
+        - cccl         # [cuda_compiler != "None"]
```

Comment:
Added CCCL. However Windows CUDA 11.8 is still using CUB, etc. from CTK. Does XGBoost have an option to specify where to get CUB, etc. from? Would like to use CCCL instead to see if that fixes the issue

Reply:
There used to be a CMake option `BUILD_WITH_CUDA_CUB` but it was removed in 2.0.0 (https://github.com/dmlc/xgboost/pull/8888).

Reply:
I see two alternatives:
1. Require CUDA 12.0 for Windows target OR
2. Add a patch to restore `BUILD_WITH_CUDA_CUB` option in XGBoost.

Reply:
`BUILD_WITH_CUDA_CUB` was there when XGBoost had cub as its submodule. We can't restore it since CCCL is a third-party package. We will have to look into how to ask CMake to prioritize the cub from CCCL over cub in CTK.

Reply:
`find_package(CCCL)` should find the conda package, in my recollection.

Reply:
About to send a patch to apply https://github.com/dmlc/xgboost/pull/10624